mlflow/models/evaluation/base.py

```diff
@@ -793,52 +793,90 @@ def evaluate(
                              column. The ``prediction`` column contains the predictions made by
                              the model. The ``target`` column contains the corresponding labels
                              to the predictions made on that row.
-                           - Dict: The second is a dictionary containing the metrics calculated
+                           - ``Dict``: The second is a dictionary containing the metrics calculated
                              by the default evaluator. The keys are the names of the metrics
                              and the values are the scalar values of the metrics. Refer to the
                              DefaultEvaluator behavior section for what metrics will be returned
                              based on the type of model (i.e. classifier or regressor).
 
                            A custom metric function can return in the following format:
 
-                           - Dict[AnyStr, Union[int, float, np.number]: a singular dictionary of
+                           - ``Dict[AnyStr, Union[int, float, np.number]``: a singular dictionary of
                              custom metrics, where the keys are the names of the metrics, and the
                              values are the scalar values of the metrics.
+                           - ``Tuple[Dict[AnyStr, Union[int,float,np.number]], Dict[AnyStr,Any]]``:
+                             a tuple of a dict containing the custom metrics, and a dict of
+                             artifacts, where the keys are the names of the artifacts, and the
+                             values are objects representing the artifacts.
+
+                           Object types that artifacts can be represented as:
+
+                           - A string uri representing the file path to the artifact. MLflow will
+                             infer the type of the artifact based on the file extension.
+                           - A string representation of a JSON object. This will be saved as a
+                             .json artifact.
+                           - ``EvaluationArtifact`` type objects. i.e. ``CsvEvaluationArtifact``,
+                             ``ImageEvaluationArtifact``.
+                           - Pandas DataFrame. This will be resolved as a CSV artifact.
+                           - Numpy array. This will be saved as a .npy artifact.
+                           - Matplotlib Figure. This will be saved as an image artifact. Note that
+                             ``matplotlib.pyplot.savefig`` is called behind the scene with default
+                             configurations. To customize, either save the figure with the desired
+                             configurations and return its file path or define customizations
+                             through environment variables in ``matplotlib.rcParams``.
+                           - Other objects will be attempted to be pickled with protocol 4.
```

Comment:
Can we not hardcode the protocol version? Rather, we should refer to the "default protocol"

Reply:
ditto: https://github.com/mlflow/mlflow/pull/5405#discussion_r821244734

Reply:
We should use the default protocol, as @BenWilson2 has discussed. Most tools use the default protocol.

Reply:
Got it.